sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationManager.kt

```diff
@@ -303,14 +307,93 @@ internal class KlaviyoLocationManager(
             return
         }
 
-        geofencingEvent.triggeringGeofences
-            ?.map { it.toKlaviyoGeofence() }
-            ?.forEach { kGeofence ->
-                // CHNL-25308 TODO enqueue API request for geofence event
-                // TODO what if the app was terminated, and the host app hasn't called `initialize` yet when we get this intent?
+        // Track all API requests we're creating
+        val requestUuids = geofencingEvent.triggeringGeofences?.map { it.toKlaviyoGeofence() }
+            ?.mapNotNull { kGeofence ->
+                if (Registry.getOrNull<State>()?.apiKey != kGeofence.companyId) {
+                    Registry.log.info("Automatically initialized Klaviyo from geofence event")
+                    Klaviyo.initialize(kGeofence.companyId, context.applicationContext)
```

Comment:
we initialize irrespective of it it's already initialized by the developer?

Reply:
Well, actually I think it should be ok. If they haven't initialized, then it means we're probably running in our own broadcast receiver process, where their full app code is not going to be launched. So we need to initialize in order to be able to make API requests. Realistically, this should only ever either be set, or be unchanging. Maybe I'll put in an error log if they're unmatched instead, and only initialize if apiKey was null.